conans/client/tools/scm.py

```diff
@@ -146,4 +157,114 @@ def get_branch(self):
+        try:
+            self.run("status")
+        except Exception:
+            raise ConanException("Not a valid git repository")
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+    file_protocol = 'file:///' if platform.system() == "Windows" else 'file://'
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache".format(command))
+
+    def checkout(self, url, revision="HEAD"):
+        output = ""
+        try:
+            self._check_svn_repo()
+        except ConanException:
+            params = " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
+            output += self.run('co "{url}" . {params}'.format(url=url, params=params))
+        else:
+            assert url.lower() == self.get_remote_url().lower(), \
+                "%s != %s" % (url, self.get_remote_url())
+            output += self.run("revert . --recursive")
+        finally:
+            output += self.update(revision=revision)
+        return output
+
+    def update(self, revision='HEAD'):
+        self._check_svn_repo()
+        return self.run("update -r {rev}".format(rev=revision))
+
+    def excluded_files(self):
+        self._check_svn_repo()
+        excluded_list = []
+        output = self.run("status --no-ignore")
+        for it in output.splitlines():
+            if it[0] == 'I':  # Only ignored files
+                filepath = it[8:].strip()
+                excluded_list.append(os.path.normpath(filepath))
+        excluded_list.append(".svn")
+        return excluded_list
+
+    def get_remote_url(self):
+        return self.run("info --show-item url").strip()
+
+    def get_qualified_remote_url(self):
+        # Return url with peg revision
+        url = self.get_remote_url()
+        revision = self.get_last_changed_revision()
+        return "{url}@{revision}".format(url=url, revision=revision)
+        
+    def is_local_repository(self):
+        url = self.get_remote_url()
+        return url.startswith(self.file_protocol) and \
+               os.path.exists(unquote(url[len(self.file_protocol):]))
+
+    def is_pristine(self):
+        # Check if working copy is pristine/consistent
+        output = self.run("status -u -r {}".format(self.get_revision()))
+        offending_columns = [0, 1, 2, 3, 4, 6, 7, 8]  # 5th column informs if the file is locked (7th is always blank)
+
+        for item in output.splitlines()[:-1]:
+            if item[0] == '?':  # Untracked file
+                continue
+            if any(item[i] != ' ' for i in offending_columns):
+                return False
+
+        return True
```

Comment:
I just played around in another project with `svn status` and think, that it will not work with this implementation. Two things, where the actual implementation will fail:
1. When using externals there will be X in the first column and also the output will come for the main project and all externals libs, so that `output.splitlines()[:-1]` will not work.
2. When using the `-u` parameter, there will be shown files that are updated on the server with an `*` . Because we want also to be able to checkout an specific revision, I think this column must also not be used.

May be we should use the `--xml` option, to better parse the files.

Reply:
I haven't check it with externals 🤦‍♂️ ... and yes, using the XML will be better, for sure.

This method, ``is_pristine``, is only used to conditionally output a warning saying that there are files modified in your working copy, so it is not a major concern (although it is a bug!). As the SCM for SVN will be experimental, we could leave it as it is (maybe removing this warning) for 1.8, start testing it (we will need more feedback, and for sure we will find more edge use cases) and prepare the XML parsing for next release. What do you think, @lasote @memsharded?

And, @climblinne, thanks a lot for your interest and your reviews and your testing. All your help is really appreciated. 🙌

Reply:
:thumbsup: Let's issue a ticket for it and try to get this feature into 1.8

Reply:
Ticket for this: #3640